scripts/mbedtls_framework/test_case.py

```diff
@@ -89,3 +111,28 @@ def write_data_file(filename: str,
             tc.write(out)
         out.write('\n# End of automatically generated file.\n')
     os.replace(tempfile, filename)
+
+def psa_or_3_6_feature_macro(psa_alg: str,
+                             domain_3_6: str) -> str:
```

Comment:
I'd prefer to use a [proper enum](https://docs.python.org/3/library/enum.html) for this parameter. Like, the class could be called `Domain_3_6` and then we have constants for `PSA`, `TLS1_3_ONLY`, `USE_PSA`, `LEGACY` so that the full name is like `Domain_3_6.USE_PSA` for example?